webapp/components/wpt-metadata.js

```diff
@@ -267,7 +276,8 @@ class WPTMetadata extends PathInfo(LoadingState(PolymerElement)) {
         curPath = curPath + '/';
       }
       const dirname = testname.substring(0, testname.length - 1);
-      return curPath.startsWith(dirname) || this.isParentDir(curPath, dirname);
+      const dirnameWithoutSlash = testname.substring(0, testname.length - 2);
+      return curPath.startsWith(dirname) || (this.isParentDir(curPath, dirname) && testResultSet.has(dirnameWithoutSlash));
```

Comment:
Is there a way to always use one of `dirname` and `dirnameWithoutSlash`? IIRC `isParentDir` accepts both?

Reply:
Correct, `isParentDir` accepts both. However, if using `dirnameWithoutSlash` for all of them, there is an edge case when `curpath` is `/abcd.html and `dirname` is `/abc`; using `dirname` for `testResultSet.has(dirnameWithoutSlash)` makes the implementation of `getDirname` more complicated. I thought it wasn't worth the work there

Reply:
But you could add `test + '/'` to `testResultSet` without having to change `getDirname`?

Reply:
Right but testResultSet has both directories and test name. To do `test + '/'`, we need to add a check when test is a testname where we don't want to add `/` (or `testResultSet.add(test)` before going into the loop) . Same goes for cases when test is `/`.  These make line 200 and 201 less elegant. But I am open to it

Reply:
Could you remind me what the purpose of `curPath.startsWith(dirname)` is? If it's not necessary, can we just keep the second part and always use `dirnameWithoutSlash`?

Reply:
The purpose is that a metadata with wildcard shows up in all children directories and I do think it is necessary

Reply:
Reading this change again, I wonder if the condition should be

```suggestion
      return (curPath.startsWith(dirname) || this.isParentDir(curPath, dirname) && testResultSet.has(dirnameWithoutSlash);
```

And we should probably make the names more meaningful (e.g. `testname` -> `metadataTestName`, `checkPath` -> `shouldShowMetadata`) and add some comments to each clause of this boolean expression.

Reply:
Sound good. The fact that you are able to see `curPath` implies `testResultSet.has(dirnameWithoutSlash)` is true but this is more rigid nevertheless